File: 2022/2022W11/PreppindataW11.py
import pandas as pd
import numpy as np

attend = pd.read_csv(r"C:\Data\PreppinData\PD Fill the Blanks challenge.csv")

attend = attend.sort_values(['Weekday', 'Lesson Time', 'Lesson Name'])
attend['Lesson Name'] = attend.groupby(['Weekday', 'Lesson Time'])['Lesson Name'].ffill()

attend = attend.sort_values(['Weekday', 'Lesson Time', 'Subject'])
attend['Subject'] = attend.groupby(['Weekday', 'Lesson Time'])['Subject'].ffill()

attend['Avg. Attendance per Subject & Lesson'] = attend.groupby(['Weekday', 'Lesson Name', 'Subject'])['Attendance'].transform(np.mean)

File: 2022/2022W11/preppindataW11.R
library(readr)
library(dplyr)

final <- read_csv("C:/Data/PreppinData/PD Fill the Blanks challenge.csv") %>%
  group_by(`Weekday`, `Lesson Time`) %>%
  mutate('Lesson Name' = max(`Lesson Name`, na.rm = TRUE),
         'Subject' = max(`Subject`, na.rm = TRUE)) %>%
  group_by(`Weekday`, `Lesson Name`, `Subject`) %>%
  mutate('Avg. Attendance per Subject & Lesson' = mean(`Attendance`)) 
  
View(final)
